hibernate-core/src/main/java/org/hibernate/id/uuid/UuidVersion6Strategy.java

```diff
@@ -98,9 +98,9 @@ public UUID generateUuid(SharedSessionContractImplementor session) {
 	private long getSequence(final long currentTimestamp) {
 		lock.lock();
 		try {
-			if ( this.currentTimestamp > currentTimestamp ) {
+			if ( this.currentTimestamp < currentTimestamp ) {
```

Comment:
Also, the spec says

>If the state was available, but the saved timestamp is less than or equal to the current timestamp, increment the clock sequence value.

Specifically **less than or equal**.

```suggestion
			if ( this.currentTimestamp <= currentTimestamp ) {
```

I'd also like to suggest using a different field name e.g. `previousTimestamp` and initialize it to `Long.MAX_VALUE`.

Reply:
Actually, I think we both might have gotten this wrong, possibly due to bad naming :)

>If the state was unavailable (e.g., non-existent or corrupted) or the timestamp is greater than the current timestamp generate a random 14-bit clock sequence value.

This part

>... or the timestamp ...

refers to the "previous timestamp". So let's initialize it to `Long.MAX_VALUE` to handle the unavailable case and rename the variable :)

```suggestion
			if ( this.previousTimestamp >= currentTimestamp ) {
```